packages/connector-jsdom/tests/get-element-final-url.ts

```diff
@@ -0,0 +1,100 @@
+
+import { test as testUtils } from '@hint/utils';
+import { Server } from '@hint/utils-create-server';
+
+import { Engine, IConnector } from 'hint';
+
+import JSDOMConnector from '../src/connector';
+
+const { generateHTMLPage } = testUtils;
+
+test(`[jsdom] getElementFinalUrl returns the same element URL if the URL is absolute`, async (t) => {
+    const baseUrl = 'http://baseurl.com/';
+    const elementUrl = 'http://absolute.com/test.png';
+    const engine = new EventEmitter2({
+        delimiter: '::',
+        maxListeners: 0,
+        wildcard: true
+    }) as Engine;
+
+    const connector: IConnector = new JSDOMConnector(engine, {});
+    const server = await Server.create({ configuration: generateHTMLPage(`<base href="${baseUrl}">`, '') });
+
+    await connector.collect(new URL(`http://localhost:${server.port}/`));
+
+    await engine.emitAsync('element::base', {
+        element: {
+            getAttribute() {
+                return baseUrl;
+            }
+        } as any,
+        resource: 'http://localhost/'
+    });
+
+    await Promise.all([connector.close(), server.stop()]);
+
+    const expected = elementUrl;
+
+    const actual = connector.getElementFinalUrl(elementUrl);
+
+    t.is(actual, expected);
+});
+
+test(`[jsdom] getElementFinalUrl returns the URL using the base.`, async (t) => {
+    const baseUrl = 'http://baseurl.com/';
+    const elementUrl = './test.png';
+    const engine = new EventEmitter2({
+        delimiter: '::',
+        maxListeners: 0,
+        wildcard: true
+    }) as Engine;
+
+    const connector: IConnector = new JSDOMConnector(engine, {});
+    const server = await Server.create({ configuration: generateHTMLPage(`<base href="${baseUrl}">`, '') });
+
+    await connector.collect(new URL(`http://localhost:${server.port}/`));
+
+    await engine.emitAsync('element::base', {
+        element: {
+            getAttribute() {
+                return baseUrl;
+            }
+        } as any,
+        resource: 'http://localhost/'
+    });
+
+    await Promise.all([connector.close(), server.stop()]);
+
+    const expected = url.resolve(baseUrl, elementUrl);
+
+    const actual = connector.getElementFinalUrl(elementUrl);
+
+    t.is(actual, expected);
+});
+
+test(`[jsdom] getElementFinalUrl returns the URL using the final URL if there is no base element.`, async (t) => {
+    const elementUrl = './test.png';
+    const engine = new EventEmitter2({
+        delimiter: '::',
+        maxListeners: 0,
+        wildcard: true
+    }) as Engine;
+
+    const connector: IConnector = new JSDOMConnector(engine, {});
+    const server = await Server.create({ configuration: generateHTMLPage('', '') });
+    const siteUrl = `http://localhost:${server.port}/`;
+
+    await connector.collect(new URL(siteUrl));
+
+    await Promise.all([connector.close(), server.stop()]);
+
+    const expected = url.resolve(siteUrl, elementUrl);
+
+    const actual = connector.getElementFinalUrl(elementUrl);
```

Comment:
Not technically a problem, but feels wrong to access data off of `connector` after calling `connector.close()`. I'd prefer to have the `await Promise.all` cleanup at the end of the test function (for this and the other new tests).